### PR TITLE
Do not display 'Show Details' button for read-only base product

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**
Closes [CXSPA-6484](https://jira.tools.sap/browse/CXSPA-6484)

Please, merge as a bug fix: 
> `fix:` for a bug fix

### CODE REVIEW

File: feature-libs/product-configurator/common/components/configure-product/configure-product.component.ts [hunk 1]
```diff
@@ -76,6 +77,38 @@ export class ConfigureProductComponent {
...
+   *
+   * @param product - product
+   * @returns - If a product a base product then returns true, otherwise false.
+   * @protected
+   */
+  protected isBaseProduct(product: Product): boolean {
+    return (
+      product.baseProduct === product.code ||
+      (!product.baseProduct && !!product.code)
+    );
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/common/components/configure-product/configure-product.component.ts]:
```suggestion
    return (
      product.baseProduct === product.code || !product.baseProduct 
    );
```
I assume `product.code` may never be `undefined`, as we query the product data using the code.

File: feature-libs/product-configurator/common/components/configure-product/configure-product.component.ts [hunk 2]
```diff
@@ -76,6 +77,38 @@ export class ConfigureProductComponent {
...
+      this.isConfiguratorTypeReadOnly(product.configuratorType) &&
+      this.isBaseProduct(product)
+    );
+  }
+
+  /**
+   * Verifies if a product is a base product.
+   *
+   * In the catalog the product is a base one when baseProduct and code are defined and equal.
+   * In the product detail page a product is a base one when only code is defined.
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/common/components/configure-product/configure-product.component.ts]: the different behavior is really strange.....

i would phrase the comment like this:
```suggestion
   * The `baseProduct` property contains the product code of the base product.
   * So if the baseProduct is undefined or if the provided code equals the code of the product itself,
   * we assume this product to be a base product.
```

**Review by Uli-Tiger — DISMISSED** [review]
In addition to the 2 comments below, do we also need to handle the case when the customer is still on older release, maybe not even using the make-to-stock feature. Will this break anything?

**Comment by Larisa-Staroverova** [comment]
Hi @Uli-Tiger,
thank you for your feedback.

> do we also need to handle the case when the customer is still on older release, maybe not even using the make-to-stock feature. Will this break anything?

If make-to-stock scenario is **not** active, then `configure` button will be displayed instead of `show details`

if make-to-stock scenario is **active**, then configurator type with READ_ONLY postfix will be returned (see `isConfiguratorTypeReadOnly` method)

To react to the older versions a feature toggle like 'productConfiguratorMakeToStockV2' should be defined that's why I asked not to merge this PR because of it.

**Comment by Uli-Tiger** [comment]
> To react to the older versions a feature toggle like 'productConfiguratorMakeToStockV2' should be defined that's why I asked not to merge this PR because of it.

agree - do we have a task for it?

**Comment by Larisa-Staroverova** [comment]
> agree - do we have a task for it?

Not yet. In first line we wanted to find a solution.
The first version of `make-to-stock` has not been merged yet.

**Review by AlexanderKuschner — APPROVED** [review]

**Review by hackergil — APPROVED** [review]

**Pull request merged**